src/profiler.cpp

```diff
@@ -660,6 +705,11 @@ Error Profiler::stop() {
     }
 
     _engine->stop();
+    if (_filter_threads != NULL) {
+        MutexLocker lock(_filtered_tids_mutex);
+        _filter_threads = NULL;
+        _filtered_tids.clear();
```

Comment:
Resetting profiler structures is typically done in `start()` rather than in `stop()`, since profiling can be later resumed (i.e. stop == pause).